tests/repository_test.py

```diff
@@ -279,14 +280,123 @@ def test_node_hook_with_npm_userconfig_set(tempdir_factory, store, tmpdir):
         test_run_a_node_hook(tempdir_factory, store)
 
 
+def _test_renv_parsing(
+    tempdir_factory,
+    store,
+    hook_id,
+    expected_hook_expr={},
+    expected_args={},
+):
+    repo_path = 'renv_hooks_repo'
+    path = make_repo(tempdir_factory, repo_path)
+    config = make_config_from_repo(path)
+    hook = _get_hook_no_install(config, store, hook_id)
+    ret = renv.cmd_from_hook(hook)
+    expected_cmd = 'Rscript'
+    expected_opts = (
+        '--no-save', '--no-restore', '--no-site-file', '--no-environ',
+    )
+    expected_path = os.path.join(
+        hook.prefix.prefix_dir, '.'.join([hook_id, 'R']),
+    )
+    expected = (
+        expected_cmd,
+        *expected_opts,
+        *(expected_hook_expr or (expected_path,)),
+        *expected_args,
+    )
+    assert ret == expected
+
+
+def test_renv_parsing_file_no_opts_no_args(tempdir_factory, store):
+    hook_id = 'parse-file-no-opts-no-args'
+    _test_renv_parsing(tempdir_factory, store, hook_id)
+
+
+def test_renv_parsing_file_opts_no_args(tempdir_factory, store):
+
+    with pytest.raises(SyntaxError, match='only valid'):
+        renv.entry_validate(['Rscript', '--no-init', '/path/to/file'])
+
+
+def test_renv_parsing_file_no_opts_args(tempdir_factory, store):
+    hook_id = 'parse-file-no-opts-args'
+    expected_args = ['--no-cache']
+    _test_renv_parsing(
+        tempdir_factory, store, hook_id, expected_args=expected_args,
+    )
+
+
+def test_renv_parsing_expr_no_opts_no_args1(tempdir_factory, store):
+    hook_id = 'parse-expr-no-opts-no-args-1'
+    _test_renv_parsing(
+        tempdir_factory, store, hook_id, expected_hook_expr=('-e', '1+1'),
+    )
+
+
+def test_renv_parsing_expr_no_opts_no_args2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='at most one'):
+        renv.entry_validate(['Rscript', '-e', '1+1', '-e', 'letters'])
+
+
+def test_renv_parsing_expr_opts_no_args2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='only valid'):
+        renv.entry_validate(
+            [
+                'Rscript', '--vanilla', '-e', '1+1', '-e', 'letters',
+            ],
+        )
+
+
+def test_renv_parsing_expr_args_in_entry2(tempdir_factory, store):
+    with pytest.raises(SyntaxError, match='at most one'):
+        renv.entry_validate(['Rscript', '-e', 'expr1', '--another-arg'])
+
+
+def test_renv_parsing_expr_non_Rscirpt(tempdir_factory, store):
+
+    with pytest.raises(SyntaxError, match='entry must start with'):
```

Comment:
I really don't like the `match=` arg and prefer to test the exact error message, you can follow this pattern:

```python
with pytest.raises(TypeHere) as excinfo:
    ...
msg, = excinfo.value.args
assert msg == ...
```